Add Behat context support

devbanana opened the pull request:
This adds support for detecting Behat context classes and marking their step definitions and hooks as used. It supports both annotation-based and attribute-based methods.

- Supports annotation-based step definitions (@Given, @When, @Then)
- Supports PHP 8 attribute-based step definitions (#[Given], #[When], #[Then])
- Supports all Behat hooks (BeforeScenario, AfterScenario, BeforeStep, AfterStep, etc.)
- Supports @Transform/#[Transform] for parameter transformations
- Auto-enables when behat/behat package is detected via Composer
- Configurable via shipmonkDeadCode.usageProviders.behat.enabled parameter

janedbal commented on tests/Rule/DeadCodeRuleTest.php:
```diff
@@ -560,8 +563,10 @@ public static function provideProviders(): Generator
         foreach (self::provideFiles() as $name => $args) {
             if (strpos($name, 'provider') !== false) {
+                $files = $args[0];
+                assert(is_string($files) || $files !== []);
```
Please remove this change upon rebase, it was already fixed by better approach in https://github.com/shipmonk-rnd/dead-code-detector/pull/262/files#diff-111f608f639db04be1c42d70e85407871fd8f3e1edfc2bbf1d7fe194390e7d6fR770

janedbal reviewed (changes requested):
Please also rebase.

devbanana commented:
Thanks @janedbal . As requested I added additional dead methods and added the feature to the README.

Yes I verified it on a personal project I am using it on along with behat. That's why I wanted to add this feature since I was getting a lot of false positives. If you need me to link to the project, I can.

Let me know if I need to make any other changes.

janedbal reviewed (approved):
Thank you!
